File: Problems/LinkedList/linkedlist_basic.py
# Node class
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

# Linked List class
class LinkedList:
    def __init__(self):
        self.head = None

    def insert_end(self, data):
        new_node = Node(data)
        if not self.head:
            self.head = new_node
            return
        temp = self.head
        while temp.next:
            temp = temp.next
        temp.next = new_node

    def print_list(self):
        temp = self.head
        while temp:
            print(temp.data, end=" -> ")
            temp = temp.next
        print("NULL")

# Driver code
if __name__ == "__main__":
    llist = LinkedList()
    llist.insert_end(1)
    llist.insert_end(2)
    llist.insert_end(3)

    print("Linked List:")
    llist.print_list()

File: Problems/LinkedList/linkedlist_basic.java

class Node {
    int data;
    Node next;
    Node(int d) {
        data = d;
        next = null;
    }
}

public class linkedlist_basic {
    Node head;

    // Insert at end
    void insertEnd(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
            return;
        }
        Node temp = head;
        while (temp.next != null) temp = temp.next;
        temp.next = newNode;
    }

    // Print list
    void printList() {
        Node temp = head;
        while (temp != null) {
            System.out.print(temp.data + " -> ");
            temp = temp.next;
        }
        System.out.println("NULL");
    }

    public static void main(String[] args) {
        linkedlist_basic list = new linkedlist_basic();
        list.insertEnd(5);
        list.insertEnd(15);
        list.insertEnd(25);

        System.out.println("Linked List:");
        list.printList();
    }
}
